src/intervals/arithmetic.jl

```diff
@@ -318,6 +346,38 @@ function radius(a::Interval)
     max(m-a.lo, a.hi-m)
 end
 
+# cancelplus and cancelminus
+@doc doc"""
+`cancelminus(a, b)` returns the unique interval `c` such that `b+c=a`.""" ->
+function cancelminus(a::Interval, b::Interval)
+    T = promote_type(eltype(a), eltype(b))
```

Comment:
I think you should be able to do this by just defining
```julia
function cancelminus{T}(a::Interval{T}, b::Interval{T})
```
and relying on the promotion rules we already have defined. Similarly for `fma` etc.